lib/ansible/executor/task_executor.py

```diff
@@ -906,38 +917,17 @@ def _get_connection(self, cvars, templar):
         # Also backwards compat call for those still using play_context
         self._play_context.set_attributes_from_plugin(connection)
 
-        if any(((connection.supports_persistence and C.USE_PERSISTENT_CONNECTIONS), connection.force_persistence)):
-            self._play_context.timeout = connection.get_option('persistent_command_timeout')
-            display.vvvv('attempting to start connection', host=self._play_context.remote_addr)
-            display.vvvv('using connection plugin %s' % connection.transport, host=self._play_context.remote_addr)
-
-            options = self._get_persistent_connection_options(connection, cvars, templar)
-            socket_path = start_connection(self._play_context, options, self._task._uuid)
-            display.vvvv('local domain socket path is %s' % socket_path, host=self._play_context.remote_addr)
-            setattr(connection, '_socket_path', socket_path)
-
         return connection
 
-    def _get_persistent_connection_options(self, connection, final_vars, templar):
-
-        option_vars = C.config.get_plugin_vars('connection', connection._load_name)
-        plugin = connection._sub_plugin
-        if plugin.get('type'):
-            option_vars.extend(C.config.get_plugin_vars(plugin['type'], plugin['name']))
-
-        options = {}
-        for k in option_vars:
-            if k in final_vars:
-                options[k] = templar.template(final_vars[k])
-
-        return options
-
-    def _set_plugin_options(self, plugin_type, variables, templar, task_keys):
-        try:
-            plugin = getattr(self._connection, '_%s' % plugin_type)
-        except AttributeError:
-            # Some plugins are assigned to private attrs, ``become`` is not
-            plugin = getattr(self._connection, plugin_type)
+    def _set_plugin_options(self, plugin_type, variables, templar, task_keys, sub=False):
+        if sub:
+            plugin = getattr(self._connection, '_sub_plugin')
+        else:
+            try:
+                plugin = getattr(self._connection, '_%s' % plugin_type)
+            except AttributeError:
+                # Some plugins are assigned to private attrs, ``become`` is not
+                plugin = getattr(self._connection, plugin_type)
```

Comment:
If I'm reading this right, I _think_ the `sub` handling is unnecessary here... all of our sub_plugins are already available as `connection.{plugin_type}`